### PR TITLE
tiup: add documents of completion, mirror clone, and list command

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Add three TiUP reference docs.

- tiup/tiup-command-completion.md
- tiup/tiup-command-mirror-clone.md
- tiup/tiup-command-list.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4868
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiup/tiup-command-completion.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- On macOS: If your bash version is lower than 4.1, run `brew install bash-completion`; otherwise, run `brew install bash-completion@2`.
+- On Linux: Use a package manager to install `bash-completion`. For example, run `yum install bash-completion` or `apt install bash-completion`.
+
+## Syntax
+
+```sh
+tiup completion <shell>
+```
+
+`<shell>` is used to set the type of shell commands. Currently, `bash` and `zsh` are supported.
```
Comment (lucklove) [inline tiup/tiup-command-completion.md:20]: 我不知道有没有shell commands 这个说法，但是我觉得好像 `the type of shell you use` 更接近一些

File: tiup/tiup-command-list.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+### --all
+
+- Displays all components. By default, TiUP does not show hidden components.
+- Data type: `BOOLEAN`
+- Default: false
+
+### --installed
+
+- 只显示已经安装的组件或版本。
```
Comment (lucklove) [inline tiup/tiup-command-list.md:27]: Do we need translate this?

File: tiup/tiup-command-mirror-clone.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+- `<target-dir>` is used to set the local path to the cloned mirror. If the path does not exist, TiUP automatically creates one.
+- If `[global version]` is specified, TiUP tries to clone all components of the specified version. If some components do not have the specified version, then clone its latest version.
+
+## Flags
+
+### -f, --full
+
+- Whether to clone the whole mirror. If this option is set, other options becomes disabled and TiUP completely clones all components of all versions from the targeted mirror.
```
Comment (lucklove) [inline tiup/tiup-command-mirror-clone.md:22]:
```suggestion
- Whether to clone the whole mirror. If this option is set, other options becomes ignored and TiUP completely clones all components of all versions from the targeted mirror.
```

File: tiup/tiup-command-list.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- Data type: `BOOLEAN`
+- Default: false
+
+## Outputs
+
+- If `[component]` is not set:
+    - If `--verbose` is specified: TiUP outputs a component information list consisting of `Name` (component name), `Installed` (installed versions), `Owner` (component owner), and `Description` (component description).
+    - If `--verbose` is not specified: TiUP outputs a component information list consisting of `Name` (component name), `Owner` (component owner), and `Description` (component description).
+- If `[component]` is set:
+    - If the specified component exists: TiUP outputs a version information list of the specified component, consisting of `Version` (version number), `Installed` (installation status),`Release` (release date), and `Platforms` (supported platforms).
```
Comment (TomShawn) [inline tiup/tiup-command-list.md:44]:
```suggestion
    - If the specified component exists: TiUP outputs a version information list of the specified component, consisting of `Version` (version number), `Installed` (installation status), `Release` (release date), and `Platforms` (supported platforms).
```

File: tiup/tiup-command-mirror-clone.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---
+title: tiup mirror clone
+---
+
+# tiup mirror clone
+
+The command `tiup mirror clone` is used to clone an existing mirror or clone some of its components to create a new mirror. The new mirror has the same components as the old one, but uses different signature keys.
```
Comment (TomShawn) [inline tiup/tiup-command-mirror-clone.md:7]: @lucklove PTAL key or keys.
```suggestion
The command `tiup mirror clone` is used to clone an existing mirror or clone some of its components to create a new mirror. The new mirror has the same components as the old one, but uses a different signature key.
```

File: tiup/tiup-command-mirror-clone.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+## Syntax
+
+```sh
+tiup mirror clone <target-dir> [global version] [flags]
+```
+
+- `<target-dir>` is used to set the local path to the cloned mirror. If the path does not exist, TiUP automatically creates one.
+- If `[global version]` is specified, TiUP tries to clone all components of the specified version. If some components do not have the specified version, then clone its latest version.
```
Comment (TomShawn) [inline tiup/tiup-command-mirror-clone.md:16]:
```suggestion
- If `[global version]` is specified, TiUP tries to clone all components of the specified version. If some components do not have the specified version, then TiUP clones its latest version.
```

File: tiup/tiup-command-mirror-clone.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+## Flags
```
Comment (TomShawn) [inline tiup/tiup-command-mirror-clone.md:18]: Options or Flags?

File: tiup/tiup-command-mirror-clone.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+### -f, --full
+
+- Whether to clone the whole mirror. If this option is set, other options becomes disabled and TiUP completely clones all components of all versions from the targeted mirror.
+- Data type: `BOOLEAN`
+- Default: false
+
+### -a, --arch
+
+- Only clone components that can run on the specified platform.
```
Comment (TomShawn) [inline tiup/tiup-command-mirror-clone.md:28]:
```suggestion
- Only clones components that can run on the specified platform.
```

File: tiup/tiup-command-mirror-clone.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+- Data type: `STRING`
+- Default: "amd64,arm64"
+
+### -o, --os
+
+- Only clone components that can run on the specified operating system.
```
Comment (TomShawn) [inline tiup/tiup-command-mirror-clone.md:34]:
```suggestion
- Only clones components that can run on the specified operating system.
```

File: tiup/tiup-command-list.md (hunk 2 above)
Comment (TomShawn) [inline tiup/tiup-command-list.md:27]:
```suggestion
```

**Review by TomShawn — COMMENTED** [review]
Minor comments

File: tiup/tiup-command-completion.md (hunk 1 above)
Comment (CharLotteiu) [inline tiup/tiup-command-completion.md:20]:
```suggestion
`<shell>` is used to set the type of shell you use. Currently, `bash` and `zsh` are supported.
```

File: tiup/tiup-command-mirror-clone.md (hunk 7 above)
Comment (CharLotteiu) [reply tiup/tiup-command-mirror-clone.md:18]: Seems like both are OK, but "options" may be a better choice according to https://superuser.com/questions/1070059/is-there-a-difference-between-a-command-line-flag-and-a-command-line-option. @lucklove Could you help us confirm this?

**Comment by TomShawn** [comment]
LGTMLGTM

File: tiup/tiup-command-mirror-clone.md (hunk 7 above)
Comment (lucklove) [reply tiup/tiup-command-mirror-clone.md:18]: options is better, I think

Comment (CharLotteiu) [reply tiup/tiup-command-mirror-clone.md]: Updated in https://github.com/pingcap/docs/pull/4965/commits/daa400580ae8a23d2fafb924b7bd21e7d8ebdd17.

**Pull request merged**